### PR TITLE
Standardize on sigmaG likelihood filter.

### PR DESCRIPTION
**Pull request by jeremykubica**
Standardizes the code to use the clipped-sigmaG filter based on likelihood with the goal of simplifying user options and the overall code base. Also moves the GPU function for filtering (which is currently unused) from filtering_kernels.cu to kernels.cu so we can remove kernels.cu.

### CODE REVIEW

**Review by DinoBektesevic — APPROVED** [review]
Always nice when more code is removed than added. Minor comments only.

File: tests/test_analysis_utils.py [hunk 1]
```diff
@@ -352,13 +321,11 @@ def test_apply_stamp_filter(self):
...
         keep = kb_post_process.load_and_filter_results(
             search,
-            {},
             self.config["lh_level"],
             chunk_size=self.config["chunk_size"],
-            filter_type="kalman",
             max_lh=self.config["max_lh"],
         )
-
+
```
[the remark this answers is not in the document]
Comment (jeremykubica) [reply tests/test_analysis_utils.py]: Good call. I forgot to run the formatter. I ran it and fixed the problems here and throughout the files.

File: src/kbmod/search/kernels.cu [hunk 2]
```diff
@@ -15,10 +15,60 @@
...
+    int tmpSortIdx;
+    for (int j = 0; j < num_values; j++) {
+        for (int k = j + 1; k < num_values; k++) {
+            if (values[idxArray[j]] > values[idxArray[k]]) {
+                tmpSortIdx = idxArray[j];
+                idxArray[j] = idxArray[k];
+                idxArray[k] = tmpSortIdx;
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (jeremykubica) [reply src/kbmod/search/kernels.cu:44]: I don't think std::sort supports this particular case where we are looking to sort an index array while keeping the values in the original order (but I could be wrong). That being said, I did implement a heap sort version of the on GPU sort and it didn't make a significant difference in runtime.

[the remark this answers is not in the document]
Comment (DinoBektesevic) [reply src/kbmod/search/kernels.cu:44]: I was a bit careless because the PR seemed pretty straight-forward. Ignore this.

**Pull request merged**